pkg/controller/operators/catalog/step_ensurer.go

```diff
@@ -116,7 +116,23 @@ func (o *StepEnsurer) EnsureServiceAccount(namespace string, sa *corev1.ServiceA
 		return
 	}
 
+	// Carrying secrets through the service account update.
+	preSa, getErr := o.kubeClient.KubernetesInterface().CoreV1().ServiceAccounts(namespace).Get(sa.Name,
+		metav1.GetOptions{})
+	if getErr != nil {
+		err = errorwrap.Wrapf(getErr, "error getting older version of service account: %s", sa.GetName())
+		return
+	}
+	// Returns error if no secret is found for the old service account. According to Kubernetes,
+	// the controller loop ensures a secret with an API token exists for each service account.
+	if len(preSa.Secrets) == 0 {
+		err = fmt.Errorf("no secretes fetched from the older version of service account: %s", sa.GetName())
+		return
+	}
```

Comment:
Could you explain why you're adding this code? If Kubernetes ensures something it seems odd that we're adding a check.

Reply:
Also - if a there are no secrets, you would be adding an empty list of secrets which should cause k8s to generate the secret (the cause of the bug you're fixing).

Reply:
you are right, we do not need this anymore.